### ISSUE TITLE
Incorrectly removing necessary forward declarations

### ISSUE THREAD

**Issue by Bothari** [include-what-you-use/include-what-you-use#331]
I've run into a pretty common and annoying bug with IWYU, where it removes a necessary forward declaration. See the attached test.cpp/.h files for an example. The output from IWYU is:

`11:14 > ./include-what-you-use.exe c:/Users/mdyckhoff/iwyu-test/test.cpp`

`c:/Users/mdyckhoff/iwyu-test/test.h should add these lines:`

`c:/Users/mdyckhoff/iwyu-test/test.h should remove these lines:`
`- class Container::Iterator;  // lines 9-9`

`The full include-list for c:/Users/mdyckhoff/iwyu-test/test.h:`
`---`

However if that line is removed, the file fails to compile, as Element is defined before Iterator:

`c:/Users/mdyckhoff/iwyu-test/test.h:13:31: error: no class named 'Iterator' in 'Container<T>'
                        friend class Container<T>::Iterator;
                                     ~~~~~~~~~~~~~~^`

I am using IWYU on Windows, in Cygwin. Version info:

`include-what-you-use 0.4 (645) based on clang version 3.6.1 (branches/release_36 238234)`

Any advice or fixes?

**Comment by Bothari** [comment]
Attached files test.cpp and test.h.

[test.zip](https://github.com/include-what-you-use/include-what-you-use/files/321065/test.zip)

**Comment by kimgr** [comment]
Thanks for the report! I can reproduce on Windows/MSVC and I agree it looks like a bug.

**Comment by kimgr** [comment]
This appears to have been fixed in IWYU 0.7!
